From 81df681fd5dafa52c5a68420adb9ce13c6585be8 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Wed, 8 Jul 2026 16:15:51 +0200
Subject: [PATCH 05/50] Plugwise: allow multiple sensors of the same kind

Assign unique appliance-scoped IDs/units for same-type readings so multiple temperatures and switch states do not collide.

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
Assisted-by: GPT-5.3-Codex

Upstream-Status: Inappropriate [WIP]
---
 hardware/AnnaThermostat.cpp | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/hardware/AnnaThermostat.cpp b/hardware/AnnaThermostat.cpp
index 0eec88190..e4fe2f874 100644
--- a/hardware/AnnaThermostat.cpp
+++ b/hardware/AnnaThermostat.cpp
@@ -538,7 +538,7 @@ void CAnnaThermostat::GetMeterDetails()
 				if (!tmpstr.empty())
 				{
 					float temperature = (float)atof(tmpstr.c_str());
-					SendTempSensor(applianceIDhex, 255, temperature, ApplianceName + " " + sname);;
+					SendTempSensor(applianceIDhex + 1, 255, temperature, ApplianceName + " " + sname);;
 				}
 			}
 			else if (sname == "return_water_temperature")
@@ -547,7 +547,7 @@ void CAnnaThermostat::GetMeterDetails()
 				if (!tmpstr.empty())
 				{
 					float temperature = (float)atof(tmpstr.c_str());
-					SendTempSensor(applianceIDhex, 255, temperature, ApplianceName + " " + sname);;
+					SendTempSensor(applianceIDhex + 2, 255, temperature, ApplianceName + " " + sname);;
 				}
 			}
 			else if (sname == "boiler_temperature")
@@ -556,7 +556,7 @@ void CAnnaThermostat::GetMeterDetails()
 				if (!tmpstr.empty())
 				{
 					float temperature = (float)atof(tmpstr.c_str());
-					SendTempSensor(applianceIDhex, 255, temperature, ApplianceName + " " + sname);;
+					SendTempSensor(applianceIDhex + 3, 255, temperature, ApplianceName + " " + sname);;
 				}
 			}
 			else if (sname == "maximum_boiler_temperature")
@@ -565,7 +565,7 @@ void CAnnaThermostat::GetMeterDetails()
 				if (!tmpstr.empty())
 				{
 					float temperature = (float)atof(tmpstr.c_str());
-					SendTempSensor(applianceIDhex, 255, temperature, ApplianceName + " " + sname);;
+					SendTempSensor(applianceIDhex + 4, 255, temperature, ApplianceName + " " + sname);;
 				}
 			}
 			else if (sname == "boiler_state")
@@ -591,11 +591,11 @@ void CAnnaThermostat::GetMeterDetails()
 					if (strcmp(tmpstr.c_str(), "on") == 0)
 					{
 
-						SendSwitch(applianceIDhex, 1, 255, true, 0, ApplianceName + " " + sname, m_Name);
+						SendSwitch(applianceIDhex, 2, 255, true, 0, ApplianceName + " " + sname, m_Name);
 					}
 					else
 					{
-						SendSwitch(applianceIDhex, 1, 255, false, 0, ApplianceName + " " + sname, m_Name);
+						SendSwitch(applianceIDhex, 2, 255, false, 0, ApplianceName + " " + sname, m_Name);
 					}
 				}
 			}
@@ -624,11 +624,11 @@ void CAnnaThermostat::GetMeterDetails()
 				{
 					if (strcmp(tmpstr.c_str(), "on") == 0)
 					{
-						SendSwitch(applianceIDhex, 1, 255, true, 0, ApplianceName + " " + sname, m_Name);
+						SendSwitch(applianceIDhex, 3, 255, true, 0, ApplianceName + " " + sname, m_Name);
 					}
 					else
 					{
-						SendSwitch(applianceIDhex, 1, 255, false, 0, ApplianceName + " " + sname, m_Name);
+						SendSwitch(applianceIDhex, 3, 255, false, 0, ApplianceName + " " + sname, m_Name);
 					}
 				}
 			}
@@ -678,7 +678,7 @@ void CAnnaThermostat::GetMeterDetails()
                 if (!tmpstr.empty())
                 {
                     float level = (float)atof(tmpstr.c_str()) * 100.F;
-                    SendPercentageSensor(applianceIDhex, 1, 255, level, ApplianceName + " " + sname);;
+					SendPercentageSensor(applianceIDhex, 2, 255, level, ApplianceName + " " + sname);;
                 }
             }
             else if (sname == "central_heater_water_pressure")
@@ -750,7 +750,7 @@ void CAnnaThermostat::GetMeterDetails()
                 if (!tmpstr.empty())
                 {
                     float temperature = (float)atof(tmpstr.c_str());
-                    SendTempSensor(applianceIDhex, 255, temperature, ApplianceName + " " + sname);;
+					SendTempSensor(applianceIDhex + 5, 255, temperature, ApplianceName + " " + sname);;
                 }
             }
         }
-- 
2.51.0

